### PR TITLE
Fix EnC issue with document source out of sync

### PR DESCRIPTION
**Pull request by drognanar**
Fixes condition that checks whether PDB and sources are out of sync.
Converts the line character position (which is relative to line start)
to a position relative to file start in order to compare it with
sourceLines.

### CODE REVIEW

**Comment by drognanar** [comment]
test prtest/mac/dbg/unit32 please

**Comment by drognanar** [comment]
test prtest/mac/dbg/unit32 please

File: src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs [hunk 1]
```diff
@@ -601,7 +601,7 @@ private void AddActiveStatements(Solution solution, ShellInterop.ENC_ACTIVE_STAT
 
                     // If the PDB is out of sync with the source we might get bad spans.
                     var sourceLines = source.Lines;
-                    if (lineSpan.End.Line >= sourceLines.Count || lineSpan.End.Character > sourceLines[sourceLines.Count - 1].EndIncludingLineBreak)
+                    if (lineSpan.End.Line >= sourceLines.Count || sourceLines.GetPosition(lineSpan.End) > sourceLines[sourceLines.Count - 1].EndIncludingLineBreak)
```
Comment (amcasey) [inline src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs]: In the old version, we were comparing a line-relative position to an absolute position?

Comment (amcasey) [inline src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs]: @tmat, is this testable?  Can we generate a PDB and then operate on modified source in a unit test?

Comment (tmat) [inline src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs]: Not easily. This code has a lot of VS dependencies. I could imagine mocking that but it'd be a significant effort.

Comment (drognanar) [inline src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs]: Regarding the issue, yes, the old version did compare line-relative to an absolute position.

**Comment by drognanar** [comment]
test prtest/mac/dbg/unit32 please

**Pull request merged**